### PR TITLE
 Hover

### PR DESCRIPTION
**Pull request by MiguelZh**

### CODE REVIEW

File: Drakhtar/Box.cpp [hunk 1]
```diff
@@ -1,21 +1,47 @@
...
-void Box::render() const {
-	GameObject::render();
+void Box::render() {
+	if (!hovered) {
+		GameObject::render();
+	}
+	else 
+	{ 
+		SDL_Rect dest{
+		pos_.getX() - size_.getX() / 2,
```
Comment (kyranet) [inline Drakhtar/Box.cpp]: This could use some padding

File: Drakhtar/Box.cpp [hunk 2]
```diff
@@ -1,21 +1,47 @@
...
+	}
 	if (content != nullptr) {
 		content->render();
 	}
 }
 
+void Box::handleEvents(SDL_Event event)
+{
+	SDL_Point p = { event.motion.x, event.motion.y };
+	if (SDL_PointInRect(&p, &this->getRect())) {
```
Comment (kyranet) [inline Drakhtar/Box.cpp]:
```cpp
hovered = SDL_PointInRect(&p, &this->getRect());
```

File: Drakhtar/Box.cpp [hunk 3]
```diff
@@ -27,3 +53,11 @@ Unit * Box::getContent() {
 void Box::setContent(Unit* object) {
 	content = object;
 }
+
+bool Box::getHovered() {
```
Comment (kyranet) [inline Drakhtar/Box.cpp]:
```suggestion
bool Box::getHovered() const {
```

**Comment by MiguelZh** [comment]
conflicts resolved

**Pull request merged**